lib/src/sql/table_type.rs

```diff
@@ -0,0 +1,20 @@
+use revision::revisioned;
+use serde::{Deserialize, Serialize};
+
+use super::Kind;
+
+#[derive(Debug, Default, Serialize, Deserialize, Hash, Clone, Eq, PartialEq, PartialOrd)]
+#[revisioned(revision = 1)]
+pub enum TableType {
+	Relation(Relation),
+	Normal,
+	#[default]
+	Any,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, Hash, Clone, Eq, PartialEq, PartialOrd)]
+#[revisioned(revision = 1)]
+pub struct Relation {
+	pub from: Option<Kind>,
+	pub to: Option<Kind>,
```

Comment:
Should we internally keep calling these `in` and `out`?

Reply:
The issue with that is in is a keyword so you'd have to do #in